src/ApaiIO/Request/Rest/Request.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/*
+ * Copyright 2016 Jan Eichhorn <exeu65@googlemail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace ApaiIO\Request\Rest;
+
+use ApaiIO\ApaiIO;
+use ApaiIO\Configuration\ConfigurationInterface;
+use ApaiIO\Operations\OperationInterface;
+use ApaiIO\Request\RequestInterface;
+use ApaiIO\Request\Util;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Basic implementation of the rest request
+ *
+ * @see    http://docs.aws.amazon.com/AWSECommerceService/2011-08-01/DG/AnatomyOfaRESTRequest.html
+ * @author Jan Eichhorn <exeu65@googlemail.com>
+ */
+class Request implements RequestInterface
+{
+    /**
+     * The requestscheme
+     *
+     * @var string
+     */
+    private $requestScheme = "http://webservices.amazon.%s/onca/xml?%s";
```

Comment:
At least if the client uses filegetcontents. I will make http as default since this is anyway a server to server communication. if a user want's to use https he can switch it in the request explicit.